server/testimony.py

```diff
@@ -0,0 +1,59 @@
+# tsuserver3, an Attorney Online server
+#
+# Copyright (C) 2016 argoneus <argoneuscze@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+class Testimony:
+    """Represents a complete group of statements to be pressed or objected to."""
+    
+    def __init__(self, title):
+        self.title = title
+        self.statements = []
+    
+    def add_statement(self, message):
+        """Add a statement and return whether successful."""
+        message = message[:14] + (1,) + message[15:]
+        self.statements.append(message)
+        return True
+    
+    def remove_statement(self, index):
+        """Remove the statement at index [index]."""
+        if index < 1 or index > len(self.statements) + 1:
+            return False
+        i = 0
+        while i < len(self.statements):
+            if i == index:
+                self.statements.remove(self.statements[i])
+                return True
+            i += 1
+        return False # shouldn't happen
+                
+    def amend_statement(self, index, message):
+        """Amend the statement at index [index] to instead contain [message]."""
+        if index < 1 or index > len(self.statements) + 1:
+            return False
+        text = message[4].split(' ')
+        message[4] = ' '.join(text[2:])
+        message[14] = 1
```

Comment:
Huh

Reply:
I forgot to comment this, will fix, but here's a rundown:
```cpp
text = message[4].split(' ')
message[4] = ' '.join(text[2:]) // amended messages start with "/amend [index]" so we remove that
message[14] = 1 // message[14] is color, and 1 is (by default) green
```